src/controls/NumericControl.tsx

```diff
@@ -0,0 +1,122 @@
+import { ControlProps, RendererProps } from "@jsonforms/core"
+import { Col, Form, InputNumber, Row, Slider } from "antd"
+import { decimalToPercentage } from "./utils"
+
+
+
+export const createNumericControl = (args: { coerceNumber: (value: number) => number; pattern?: string }) => {
+  return function NumericControl({
+    data,
+    handleChange,
+    path,
+    required,
+    label,
+    visible,
+    id,
+    schema,
+    uischema,
+  }: ControlProps & RendererProps) {
+    const arialLabelWithFallback = label || schema.description || "Value"
+    const isRequired = required || uischema.options?.required as boolean
```

Comment:
I think this reads better if you type the options object separately, e.g.
```tsx
const options: NumberControlOptions = uischema.options as NumberControlOptions
```
and then casts like
```
 const addonAfter = uischema.options?.addonAfter as string | undefined
```
are not necessary

Reply:
Why do we need to add `NumberControlOptions` on that line twice?

Reply:
Ah, yeah we don't!